kube-runtime/src/utils/stream_subscribable.rs

```diff
@@ -0,0 +1,58 @@
+use core::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+use futures::{stream, Stream, TryStream};
+use pin_project::pin_project;
+use tokio::sync::broadcast;
+
+#[pin_project]
+/// todo - docs
+#[must_use = "streams do nothing unless polled"]
+pub struct StreamSubscribable<S>
+where
+    S: TryStream,
+{
+    #[pin]
+    stream: S,
+    sender: broadcast::Sender<S::Ok>,
+}
```

Comment:
I do like the setup of this now. It's fairly orthogonal to the rest of this PR and should have wide usability outside of the more ambitious scope of this PR, so it might be worth cleaning this file up (plus the watch_ext callee) in an own PR so we can reduce the scope here.

There's only tests and docs (incl a doc example). Tests can follow other WatchStreamExt setup and check some reasonable edge cases; more than 1 auxiliary subscriber,  and verify that sending an Err propagates the err everywhere, etc.